centreon/src/Core/Dashboard/Playlist/Infrastructure/API/ListPlaylistShares/ListPlaylistSharesPresenter.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+/*
+ * Copyright 2005 - 2023 Centreon (https://www.centreon.com/)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+declare(strict_types=1);
+
+namespace Core\Dashboard\Playlist\Infrastructure\API\ListPlaylistShares;
+
+use Core\Application\Common\UseCase\AbstractPresenter;
+use Core\Application\Common\UseCase\ResponseStatusInterface;
+use Core\Dashboard\Playlist\Application\UseCase\ListPlaylistShares\ListPlaylistSharesPresenterInterface;
+use Core\Dashboard\Playlist\Application\UseCase\ListPlaylistShares\ListPlaylistSharesResponse;
+
+class ListPlaylistSharesPresenter extends AbstractPresenter implements ListPlaylistSharesPresenterInterface
+{
+    public function presentResponse(ListPlaylistSharesResponse|ResponseStatusInterface $response): void
+    {
+        if ($response instanceof ResponseStatusInterface) {
+            $this->setResponseStatus($response);
+        } else {
+            $this->present([
+                'contacts' => $response->contacts,
+                'contact_groups' => $response->contactGroups,
```

Comment:
I would have liked an array_map on this instead of using the response arrays directly but ok

Reply:
Generally i do the array_map on the useCase while creating the response

```
        $response->contactGroups = array_map(function (PlaylistContactGroupShare $contactShare): array {
            return [
                'id' => $contactShare->getContactGroupId(),
                'name' => $contactShare->getContactGroupName(),
                'role' => $contactShare->getRole(),
            ];
        }, $shares->getPlaylistContactGroupShares());
        ```